helm/cert-manager-app/charts/crd-install-cleanup/templates/job.yaml

```diff
@@ -5,7 +5,7 @@ metadata:
   namespace: {{ .Release.Namespace }}
   annotations:
     "helm.sh/hook": "post-install,post-upgrade"
-    "helm.sh/hook-delete-policy": "hook-succeeded,hook-failed"
+    "helm.sh/hook-delete-policy": "before-hook-creation,hook-succeeded,hook-failed"
```

Comment:
I added `before-hook-creation` to recommended `hook-succeeded,hook-failed` to be able to automatically deploy this fix to all affected installations with no manual intervention. 

It is not needed in the long run and can be discarded once this fix is released.

Reply:
i'd be inclined to leave it in the chart anyway as hooks can be flakey as hell for no reason - i've often seen them take several attempts to succeed.

Reply:
Yes I agree better to keep this in. Should make it more reliable since it can take multiple attempts for upgrades to succeed.